### PR TITLE
Convert OverlayCallbackImpl to Kotlin

### PR DESCRIPTION
**Pull request by Goooler**
## Type of change

✅ General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:x: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import com.google.android.libraries.launcherclient.LauncherClientService
+import com.google.android.libraries.launcherclient.StaticInteger
+import com.patrykmichalik.opto.core.firstBlocking
+
+/**
+ * Implements [LauncherOverlay] and passes all the corresponding events to [LauncherClient],
+ * see [LauncherClientService.setClient].
+ *
+ * Implements [LauncherClientCallbacks] and sends all the corresponding callbacks to [ ].
+ */
```
Comment (yasanglass) [inline lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt]: The document is missing the reference to `Launcher` like the original java file.

```suggestion
import com.android.launcher3.Launcher
import com.android.launcher3.Utilities
import com.android.systemui.plugins.shared.LauncherOverlayManager
import com.android.systemui.plugins.shared.LauncherOverlayManager.LauncherOverlay
import com.android.systemui.plugins.shared.LauncherOverlayManager.LauncherOverlayCallbacks
import com.google.android.libraries.launcherclient.ISerializableScrollCallback
import com.google.android.libraries.launcherclient.LauncherClient
import com.google.android.libraries.launcherclient.LauncherClientCallbacks
import com.google.android.libraries.launcherclient.LauncherClientService
import com.google.android.libraries.launcherclient.StaticInteger
import com.patrykmichalik.opto.core.firstBlocking

/**
 * Implements [LauncherOverlay] and passes all the corresponding events to [LauncherClient],
 * see [LauncherClientService.setClient].
 *
 * Implements [LauncherClientCallbacks] and sends all the corresponding callbacks to [Launcher].
 */
```

File: lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    override fun hideOverlay(animate: Boolean) {
+        mClient.hideOverlay(animate)
+    }
+
+    override fun hideOverlay(duration: Int) {
+        mClient.hideOverlay(duration)
+    }
+
```
Comment (yasanglass) [inline lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt:66]:
```suggestion

    override fun startSearch(config: ByteArray?, extras: Bundle?): Boolean {
        return false
    }

    override fun onActivityCreated(activity: Activity, bundle: Bundle?) {
        // Not called
    }

```

These two overrides are missing from the new Kotlin file, I know they match the super methods but as far as I know, they should be still overridden here.

**Review by yasanglass — CHANGES_REQUESTED** [review]
Just a few small changes.

File: lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt (hunk 2 above)
Comment (Goooler) [reply lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt:66]: be1bdf847d6593257490dc28d8c63f9dfcf36b87

Comment (Goooler) [reply lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt:66]: Nice catch!

File: lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt [hunk 3]
```diff
@@ -81,8 +79,6 @@ class OverlayCallbackImpl(private val mLauncher: LawnchairLauncher) : LauncherOv
         mClient.onStop()
     }
 
-    override fun onActivitySaveInstanceState(activity: Activity, bundle: Bundle?) = Unit
-
```
Comment (Goooler) [inline lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt]: Does this need to be added back?

Comment (yasanglass) [reply lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt]: Yeah all the overrides should be on the new file as well, I missed this one on my review.

Comment (Goooler) [reply lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt]: d0fbd9dc7023b2de24020c91977636fa8be6c34e

Comment (Goooler) [reply lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt]: I know why I removed this, the reason:

```
e: /home/runner/work/lawnchair/lawnchair/lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt: (87, 5): 'onActivitySaveInstanceState' overrides nothing

FAILURE: Build completed with 2 failures.

> Task :compileLawnWithQuickstepDebugKotlin FAILED
```

Comment (Goooler) [reply lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt]: 3c873a3dd77cdeb526652f2be652b7209c46f1ce

Comment (yasanglass) [reply lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt]: Nice, yeah it looks like the wrong param was the issue.

**Review by yasanglass — APPROVED** [review]

**Pull request merged**